numpy/_core/src/multiarray/ctors.c

```diff
@@ -1615,6 +1622,10 @@ PyArray_FromAny_int(PyObject *op, PyArray_Descr *in_descr,
         assert(cache->converted_obj == op);
         arr = (PyArrayObject *)(cache->arr_or_sequence);
         /* we may need to cast or assert flags (e.g. copy) */
+        if (was_copied_by__array__ == 1 && flags & NPY_ARRAY_ENSURECOPY) {
```

Comment:
```suggestion
        if (was_copied_by__array__ == 1) {
```
You don't need to test this, it is OK to unset the flag if it isn't set.  (to me that is clearer, but maybe it's me)

Reply:
Done!